tutorial/source/tensor_shapes.ipynb

```diff
@@ -0,0 +1,292 @@
+    "  - use negative indices like `x.sum(-1)` rather than `x.sum(2)`\n",
+    "  - use ellipsis notation like `pixel = image[..., i, j]`\n",
+    "  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "import pyro\n",
+    "from pyro.distributions import Bernoulli, MultivariateNormal, Normal"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Distributions shapes: `batch_shape` and `event_shape`\n",
+    "\n",
+    "The simplest distribution shape is a single univariate distribution."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "d = Bernoulli(0.5)\n",
+    "assert d.batch_shape == ()\n",
+    "assert d.event_shape == ()\n",
+    "x = d.sample()\n",
+    "assert x.shape == ()\n",
+    "assert d.log_prob(x).shape == ()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Distributions can be batched by passing in batched parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "d = Bernoulli(0.5 * torch.ones(3,4))\n",
+    "assert d.batch_shape == (3, 4)\n",
+    "assert d.event_shape == ()\n",
+    "x = d.sample()\n",
+    "assert x.shape == (3, 4)\n",
+    "assert d.log_prob(x).shape == (3, 4)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Another way to batch distributions is via the `.reshape()` method. This only works if \n",
+    "parameters are identical along the leftmost dimensions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "d = Bernoulli(torch.tensor([0.1, 0.2, 0.3, 0.4])).reshape([3])\n",
+    "assert d.batch_shape == (3, 4)\n",
+    "assert d.event_shape == ()\n",
+    "x = d.sample()\n",
+    "assert x.shape == (3, 4)\n",
+    "assert d.log_prob(x).shape == (3, 4)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Multivariate distributions have nonempty `.event_shape`. For these distributions, the shapes of `.sample()` and `.log_prob(x)` differ:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "d = MultivariateNormal(torch.zeros(3), torch.eye(3, 3))\n",
+    "assert d.batch_shape == ()\n",
+    "assert d.event_shape == (3,)\n",
+    "x = d.sample()\n",
+    "assert x.shape == (3,)            # == batch_shape + event_shape\n",
+    "assert d.log_prob(x).shape == ()  # == batch_shape"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In Pyro you can treat a univariate distribution as multivariate by calling the `.reshape(extra_event_dims=_)` property."
```

Comment:
separate this and the next section from the previous one with a header like "Declaring dependence" or something